core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -228,7 +228,10 @@ private void configureConnectors() {
     for (NamedURI listener : listeners) {
       if (listener.getUri().getScheme().equals("https")) {
         if (httpConfiguration.getCustomizer(SecureRequestCustomizer.class) == null) {
-          httpConfiguration.addCustomizer(new SecureRequestCustomizer());
+          SecureRequestCustomizer secureRequestCustomizer = new SecureRequestCustomizer();
+          // Explicitly making sure that SNI is checked against Host in HTTP request
+          secureRequestCustomizer.setSniHostCheck(true);
```

Comment:
Wouldn't it be better to write the `assert` instead of explicit check as the config is enabled by default?

```suggestion
          assert secureRequestCustomizer.isSniHostCheck();
```

Reply:
@apoorvmittal10 that also works, however I'm not sure if we're allowed to use assert in source (not tests) in Confluent.

Reply:
I can see several examples of that in [ce-kafka](https://github.com/search?q=repo%3Aconfluentinc%2Fce-kafka%20%22assert%20%22&type=code). Moreover it's java `assert` keyword unlike any `Assertion` from test library. I think it should be safe to use that.
cc: @AndrewJSchofield

Reply:
Given that precedent, I would say it's OK.